dev/tools/test-migration.py

```
@@ -233,30 +241,36 @@ def install_v1alpha1(method, version):
-                    if (member.name.endswith("/helm") or member.name.endswith("/helm/")) and member.isdir():
-                        helm_src_dir = member.name
-                        break
-
-                if not helm_src_dir:
-                    # Guess default structure
-                    helm_src_dir = f"agent-sandbox-{helm_version}/helm/"
-                tar.extractall(path=temp_dir)
+            if os.path.exists(local_tarball):
+                print(f"Extracting local Helm chart from {local_tarball}...")
+                with tarfile.open(local_tarball, "r:gz") as tar:
+                    tar.extractall(path=temp_dir)
```

Ruff (0.15.17)

[error] Uses of `tarfile.extractall()`

(S202)